dijet/tasks/jer.py

```diff
@@ -60,23 +60,28 @@ def output(self) -> dict[law.FileSystemTarget]:
         return outp
 
     def run(self):
-        widths = self.load_alpha()
+        widths = self.load_widths()
 
         # ### Now JER SM Data
+        jer = widths["fits"].copy()
+        view = jer.view()
 
-        jer_sm = widths["sm"]["fits"][0][:, :] * np.sqrt(2)
+        # Get index of method to change methods individually
+        # hist.view() only works if the full histgram is taken w.o. selecting the categories before hand
+        category_id = {"sm": 1, "fe": 2}
+        # Get list of categories with the correct order
+        categories = list(jer.axes["category"])
+        index_methods = {m: categories.index(category_id[m]) for m in category_id}
+
+        # calcuate jer for standard method
+        view.value[index_methods["sm"]] = view.value[index_methods["sm"], :, :] * np.sqrt(2)
 
         # TODO: Define eta bin in config
         # NOTE: weighting number by appearence in eta regions
-        jer_ref = np.mean(jer_sm[:5, :], axis=0).reshape(1, -1)
-        jer_fe = np.sqrt(4 * widths["fe"]["fits"][0][:, :]**2 - jer_ref**2)
+        jer_ref = np.mean(view.value[index_methods["sm"], :5, 0], axis=0).reshape(1, -1)
+        view.value[index_methods["fe"]] = np.sqrt(4 * view.value[index_methods["fe"], :, :]**2 - jer_ref**2)
 
         results_jers = {
-            "sm": {
-                "jers": jer_sm,   
-            },
-            "fe": {
-                "jers": jer_fe,  
-            },
+            "jer": jer,
```

Comment:
The `run` method here is very hard to read due to the variable naming scheme, the complicated indexing expressions, and use of views to update the `jer` object in-place. While some of the complexity here is unavoidable, I think the code can be streamlined further to make it more clear to the uninitiated reader. To that end, I would suggest working with local variables for storing the different steps of the computation and filling the histogram from those at the end.

Here's an implementation suggestion to illustrate what I mean. I haven't tested it, but it could serve as a basis for cleaning up the code here somewhat:
```
    def run(self):
        # load extrapolation results
        results_extrapolation = self.load_extrapolation()

        # get extrapolated distribution widths
        h_widths = results_extrapolation["intercepts"]

        # get index on `category` axis corresponding to
        # the two computation methods
        category_id = {"sm": 1, "fe": 2}
        categories = list(h_widths.axes["category"])
        index_methods = {m: categories.index(category_id[m]) for m in category_id}

        # calcuate JER for standard method
        jer_sm_val = widths[index_methods["sm"], :, :].value * np.sqrt(2)
        jer_sm_err = np.sqrt(h_widths[index_methods["sm"], :, :].variance) * np.sqrt(2)

        # average over first few eta bins to get
        # reference JER for forward method
        jer_ref = np.mean(jer_sm_val[:5, :], axis=0, keepdims=True)

        # calculate JER for forward extension method
        jer_fe_val = np.sqrt(4 * h_widths[index_methods["fe"], :, :].value**2 - jer_ref**2)
        # jer_fe_err = ???   # TODO: calculate via error propagation

        # create output histogram and view for filling
        h_jer = h_widths.copy()
        v_jer = h_jer.view()

        # write JER values to output histogram
        v_jer[index_methods["sm"], :, :].value = jer_sm_val
        v_jer[index_methods["sm"], :, :].variance = jer_sm_err**2
        v_jer[index_methods["fe"], :, :].value = jer_fe_val
        v_jer[index_methods["fe"], :, :].variance = jer_fe_err**2

        results_jers = {
            "jer": h_jer,
        }
        self.output()["jers"].dump(results_jers, formatter="pickle")
```